### ISSUE TITLE
[a11y]: Accessibility Checker finds violation on ClickableTile

### ISSUE THREAD

**Issue by miriam-g** [carbon-design-system/carbon#18704]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

"@carbon/react": "^1.71.1"

### React version

v17.0.2

### Automated testing tool and ruleset

IBM Equal Accessibility Checker - Latest deployment

### Description

We are using ClickableTiles in our UI. When scanning the page with the IBM Accessibility Checker (guidelines IBM Accessibility 7.3) we get the following violations:
`The tabbable element does not have a valid widget role`
Using role="button" fixes the issue, but for accessibility reasons we are asked to prefer native HTML elements over defining roles (due to [the first ARIA rule](https://www.w3.org/TR/aria-in-html/#rule1)). Are you planning to fix this on your side?

### Reproduction/example

-

### Steps to reproduce

Run the accessibility checker with accessibility guidelines on a page that uses the ClickableTile without defining a role.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alisonjoseph** [comment]
Can you provide a reproduction in [Stackblitz](https://stackblitz.com/run?file=src%2FApp.jsx) on how you are using this component, I don't see any a11y violations in the Storybook.

**Comment by miriam-g** [comment]
This is how we use it: https://stackblitz.com/edit/github-if9b6zjg?file=src%2FApp.jsx&preset=node=

**Comment by ariellalgilmore** [comment]
Can you try adding a `href`? or maybe just using `Tile`.  Looks like you are using a `ClickableTile` with no type of action.

**Comment by miriam-g** [comment]
I did not add our whole code in the Stackblitz, but we use actions (no URLs). So I need a clickable tile.

**Comment by ariellalgilmore** [comment]
Can you provide an example that more closely represents the ClickableTiles you have?

**Comment by miriam-g** [comment]
You can already reproduce the issue with my stackblitz:
Please open the Preview in a separate Tab and run the IBM Accessibility Checker on this page. Then you see the mentioned violations.

**Comment by thyhmdo** [comment]
@miriam-g Hi there, we’ll reach out to an accessibility expert for clarification. In the meantime, could you let us know if this is blocking you from achieving the intended design or user experience?

**Comment by miriam-g** [comment]
Thank you @thyhmdo, no, it is not blocking me.

**Comment by mbgower** [comment]
in regard to this code, 

![Image](https://github.com/user-attachments/assets/a3da5b44-865c-4dad-8e7f-e374c616c62a)


 the clickableTile ‘elements’ say type="button" but that is not a role. If instead it was role="button" the error would likely go away.

**Comment by 2nikhiltom** [comment]
Hey @miriam-g 

> the clickableTile ‘elements’ say type="button" but that is not a role. If instead it was role="button" the error should go away.

This does fix the issue : fix example  [stackblitz](https://stackblitz.com/edit/github-if9b6zjg-vdtxyopa?file=src%2FApp.jsx&preset=node=) 
Could you confirm from your end ?

**Comment by 2nikhiltom** [comment]
Hey @miriam-g 

> Hey @miriam-g
> 
> the clickableTile ‘elements’ say type="button" but that is not a role. If instead it was role="button" the error should go away.
> 
> This does fix the issue : fix example [stackblitz](https://stackblitz.com/edit/github-if9b6zjg-vdtxyopa?file=src%2FApp.jsx&preset=node=)
> Could you confirm from your end ?

I am closing the issue with the above comments, happy to re-open if you need anything else

**Comment by miriam-g** [comment]
Yes, confirmed. That's what I also mentioned in my description. 
Thanks for looking into it.
